src/main/python/run_trecqa.py

```diff
@@ -5,219 +5,294 @@
-      ps_split = ps.split('\t')
-      candidate_passages.append(ps_split[0])
-
-    # TODO: for processing input data. Ideally these settings need to come in as program arguments
-    # NOTE: the best model for TrecQA keeps punctuation and keeps dash-words
-    flags = {
-      "punctuation": "", # ignoring for now  you can {keep|remove} punctuation
-      "dash_words": "" # ignoring for now. you can {keep|split} words-with-hyphens
-    }
-    answers_list = qa_DL_model.rerank_candidate_answers(question, candidate_passages, idf_json, flags)
-    sorted_answers = sorted(answers_list, key=lambda x: x[0], reverse=True)
-
-    for score, candidate in sorted_answers:
-      tokens = TreebankWordTokenizer().tokenize(candidate.lower().split("\t")[0])
-      tokeninzed_answer.append((tokens, score))
-    return tokeninzed_answer
-
-  elif model_choice == 'idf':
-    for candidate in candidate_passages_scores:
-      candidate_sent, score = candidate.lower().split("\t")
-      tokens = TreebankWordTokenizer().tokenize(candidate_sent.lower())
-      tokeninzed_answer.append((tokens, score))
-    return tokeninzed_answer
+    instance = None
+    def __init__(self, model_choice, index_path, w2v_cache, qa_model_file):
+        if model_choice == "sm":
+            QAModel.instance = SMModelBridge(qa_model_file, w2v_cache, index_path)
+
+def get_answers(pyserini, question, h0, h1, model_choice, index_path, w2v_cache="", qa_model_file=""):
+    candidate_passages_scores = pyserini.ranked_passages(question, h0, h1)
+    idf_json = pyserini.get_term_idf_json()
+    candidate_passages = []
+    tokeninzed_answer = []
+
+    if model_choice == 'sm':
+
+        qa_DL_model = QAModel(model_choice, index_path, w2v_cache, qa_model_file).instance
+
+        for ps in candidate_passages_scores:
+            ps_split = ps.split('\t')
+            candidate_passages.append(ps_split[0])
+
+        # TODO: for processing input data. Ideally these settings need to come in as program arguments
+        # NOTE: the best model for TrecQA keeps punctuation and keeps dash-words
+        flags = {
+          "punctuation": "", # ignoring for now  you can {keep|remove} punctuation
+          "dash_words": "" # ignoring for now. you can {keep|split} words-with-hyphens
+        }
+        answers_list = qa_DL_model.rerank_candidate_answers(question, candidate_passages, idf_json, flags)
+        sorted_answers = sorted(answers_list, key=lambda x: x[0], reverse=True)
+
+        for score, candidate in sorted_answers:
+            tokens = TreebankWordTokenizer().tokenize(candidate.lower().split("\t")[0])
+            tokeninzed_answer.append((tokens, score))
+        return tokeninzed_answer
+
+    elif model_choice == 'idf':
+        for candidate in candidate_passages_scores:
+            candidate_sent, score = candidate.lower().split("\t")
+            tokens = TreebankWordTokenizer().tokenize(candidate_sent.lower())
+            tokeninzed_answer.append((tokens, score))
```

Comment:
These few lines look suspiciously similar to the last few lines of the prior `if` condition ... perhaps some refactoring would make them only written once.